### PR TITLE
LP v2: fix integration tests

### PR DESCRIPTION
**Pull request by wischli**
# Description

* Fixes more ITs
  * The last 5 ignored ones are mysterious - will dig deeper into those after updating Dev chain with our current LPv2 PoC
* Updates Solidity submodule to latest state

# Checklist:

- [x] I have added Rust doc comments to structs, enums, traits and functions
- [x] I have made corresponding changes to the documentation
- [x] I have performed a self-review of my code
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: runtime/integration-tests/src/cases/lp/setup_evm.rs [hunk 1]
```diff
@@ -0,0 +1,578 @@
+// Copyright 2024 Centrifuge Foundation (centrifuge.io).
```
Comment (wischli) [inline runtime/integration-tests/src/cases/lp/setup_evm.rs:1]: NOTE: Code moved and improved from `lp/mod.rs` following Solidity structure

File: runtime/integration-tests/src/cases/lp/setup_lp.rs [hunk 2]
```diff
@@ -0,0 +1,966 @@
+// Copyright 2024 Centrifuge Foundation (centrifuge.io).
```
Comment (wischli) [inline runtime/integration-tests/src/cases/lp/setup_lp.rs:1]: NOTE: Code moved and improved from lp/mod.rs following Solidity structure

File: runtime/integration-tests/src/cases/lp/utils.rs [hunk 3]
```diff
@@ -0,0 +1,310 @@
+// Copyright 2024 Centrifuge Foundation (centrifuge.io).
```
Comment (wischli) [inline runtime/integration-tests/src/cases/lp/utils.rs:1]: NOTE: Code moved and improved from lp/mod.rs

File: runtime/integration-tests/src/cases/lp/transfers.rs [hunk 4]
```diff
@@ -244,10 +264,12 @@ fn transfer_tranche_tokens_domain_to_local_to_domain<T: Runtime>() {
 			Keyring::TrancheInvestor(1),
 			sp_core::U256::zero(),
 			names::POOL_MANAGER,
-			"transferTranchesToEVM",
+			"transferTrancheTokens",
 			Some(&[
 				Token::Uint(POOL_A.into()),
 				Token::FixedBytes(pool_a_tranche_1_id::<T>().into()),
+				// Represents Solidity enum Domain.EVM
+				Token::Uint(1.into()),
```
Comment (lemunozm) [inline runtime/integration-tests/src/cases/lp/transfers.rs:272]: Super NIT. Using a constant.

File: runtime/integration-tests/src/cases/lp/transfers.rs [hunk 5]
```diff
@@ -321,6 +344,10 @@ fn transfer_tranche_tokens_domain_to_local<T: Runtime>() {
 			Some(&[
 				Token::Uint(POOL_A.into()),
 				Token::FixedBytes(pool_a_tranche_1_id::<T>().into()),
+				// Represents Solidity enum Domain.Centrifuge
+				Token::Uint(0.into()),
+				// Represents Centrifuge Chain id which is 0
+				Token::Uint(0.into()),
```
Comment (lemunozm) [inline runtime/integration-tests/src/cases/lp/transfers.rs:350]: Super NIT. Same here

File: runtime/integration-tests/src/cases/lp/investments.rs [hunk 6]
```diff
@@ -156,7 +155,7 @@ mod with_pool_currency {
 	use super::{utils, *};
 	use crate::cases::lp::utils as lp_utils;
 
-	#[test_runtimes([development], ignore = "solidity mismatch")]
+	#[test_runtimes([development])]
```
Comment (lemunozm) [inline runtime/integration-tests/src/cases/lp/investments.rs:158]: FYI: There is a TODO 10 lines above.

**Review by lemunozm — APPROVED** [review]
Looks great William!

I'm not an expert at all in the EVM part, so I could not notice nothing wrong on it.

Added some super NIT things that can be skipped perfectly.

**Comment by lemunozm** [comment]
Let's wait to see everything green before merging, if possible 🙏🏻

File: runtime/integration-tests/src/cases/lp/investments.rs (hunk 6 above)
Comment (wischli) [reply runtime/integration-tests/src/cases/lp/investments.rs:158]: Thanks, that's from @mustermeiszer and not related to the LPv2 refactor I have applied.

File: runtime/integration-tests/src/cases/lp/transfers.rs (hunk 4 above)
Comment (wischli) [reply runtime/integration-tests/src/cases/lp/transfers.rs]: Thanks! [d7ab181](https://github.com/centrifuge/centrifuge-chain/pull/1915/commits/d7ab18103e001115bdf64fc0fb8d66e4cc180115)

**Comment by wischli** [comment]
@mustermeiszer @lemunozm @hieronx FYI: I am taking this state for our Dev deployment today. It reflects the latest state of our branch `feat/lp-v2`. Once the CI is green, we will merge this branch into our feature one

File: runtime/integration-tests/src/cases/lp/mod.rs [hunk 7]
```diff
@@ -93,6 +93,14 @@ pub const EVM_DOMAIN_CHAIN_ID: u64 = 1;
 
 pub const EVM_DOMAIN: Domain = Domain::EVM(EVM_DOMAIN_CHAIN_ID);
 
+/// Represents Solidity enum Domain.Centrifuge
+pub const DOMAIN_CENTRIFUGE: u8 = 0;
+
+/// Represents Solidity enum Domain.Evm
+pub const DOMAIN_EVM: u8 = 1;
+/// Represents Centrifuge Chain id which is 0
+pub const CENTRIFUGE_CHAIN_ID: u8 = 0;
```
Comment (lemunozm) [inline runtime/integration-tests/src/cases/lp/mod.rs:102]: ❤️

**Pull request merged**